Security/Firewall/OAuthListener.php

```diff
@@ -44,13 +44,13 @@ class OAuthListener implements ListenerInterface
     protected $serverService;
 
     /**
-     * @param \Symfony\Component\Security\Core\SecurityContextInterface                      $securityContext       The security context.
-     * @param \Symfony\Component\Security\Core\Authentication\AuthenticationManagerInterface $authenticationManager The authentication manager.
-     * @param \OAuth2\OAuth2                                                                 $serverService
+     * @param \Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface   $tokenStorage          The token storage.
+     * @param \Symfony\Component\Security\Core\Authentication\AuthenticationManagerInterface        $authenticationManager The authentication manager.
+     * @param \OAuth2\OAuth2                                                                        $serverService
      */
-    public function __construct(SecurityContextInterface $securityContext, AuthenticationManagerInterface $authenticationManager, OAuth2 $serverService)
+    public function __construct(TokenStorageInterface $tokenStorage, AuthenticationManagerInterface $authenticationManager, OAuth2 $serverService)
```

Comment:
instead of supporting only the new API, please support both the old and the new one, so that we can keep symfony 2.3 support until the end of its LTS.